jupyterlab_gitsync/src/service/notebook_file.ts

```diff
@@ -41,55 +50,100 @@ export class NotebookFile implements IFile {
     return this._dirtyState;
   }
 
+  get activeCell() {
+    return this.content.activeCell;
+  }
+
+  get activeCellIndex(): number {
+    return this.content.activeCellIndex;
+  }
+
   async save() {
     try{
-      await this.context.save();
+      const content = (this.content.model as NotebookModel).toJSON();
+      await this._saveFile(content);
+      this.resolver.addVersion(content, 'base');
     } catch (error) {
       console.warn(error);
     }
   }
 
   async reload() {
+    await this._getRemoteVersion();
     this._getLocalVersion();
-    this._getRemoteVersion();
+    this._getEditorView();
+    const merged = await this.resolver.mergeVersions();
+    if (merged) { await this._displayText(merged); }
+  }
+
+  private async _displayText(merged) {
+    await this._saveFile(merged)
     await this.context.revert();
+    this._setEditorView();
+  }
+
+  private async _saveFile(content: any){
+    const options = {
+      content: content,
+      format: 'json' as Contents.FileFormat,
+      path: this.path,
+      type: 'notebook' as Contents.ContentType,
+    }
+    await fs.save(this.path, options);
+  }
+
+  private async _getInitVersion() {
+    const contents = await fs.get(this.path);
+    this.resolver.addVersion(contents.content, 'base');
+    this.resolver.addVersion(contents.content, 'remote');
+    this.resolver.addVersion(contents.content, 'local');
   }
 
   private async _getRemoteVersion() {
-    // TO DO (ashleyswang): change to get raw string of notebook file
-    const contents = await fs.get(this.path, {content: false});
-    console.log(contents.content);
+    try{
+      const contents = await fs.get(this.path);
+      this.resolver.addVersion(contents.content, 'remote');
+    } catch (error) {
+      console.warn(error);
```

Comment:
you probably can build a centralized place to display all the error messages. maybe together with the status indicator? so when the status is red, users get some error messages.

Reply:
I'm thinking of displaying the errors somewhere on the UI (maybe clicking the status indicator will show a panel with a log of all service operations, ie. save/sync/etc.), but this will most likely be included in a separate PR.